### ISSUE TITLE
Crashing on Swift 3

### ISSUE THREAD

**Issue by johnpaulmanoza** [matthewpalmer/Locksmith#156]
Hi Im having a trouble loading the keychain data. App crash when loadDataForUserAccount is called. xcode stops here Locksmith.swift (Line 84)

![screen shot 2016-10-26 at 2 07 16 pm](https://cloud.githubusercontent.com/assets/2209713/19715122/a911b52c-9b85-11e6-979e-9c755556ca27.png)

Locksmith 3.0.0
Xcode 8.0

**Comment by rlam3** [comment]
@johnpaulmanoza  were you able to resolve this problem? I'm having a similar issue as well...

**Comment by johnnyoin** [comment]
I have this issue as well but the app doesn't crash. Locksmith complains about a duplicate but still does the job.

**Comment by Micke5** [comment]
Any progress on this issue? I'm having the same problem :(

**Comment by rlam3** [comment]
@johnnyoin Were you able to resolve this issue?

**Comment by johnnyoin** [comment]
@rlam3 I didn't have to since the app doesn't crash in my case. It stops on the Swift Error breakpoint but still works properly.

**Comment by iwasrobbed** [comment]
The whole point of `throw` is for devs to use `try` and catch those errors so it won't crash. If you're doing `try!` instead of `try` then you're causing the crash, not Locksmith.

**Comment by rlam3** [comment]
@iwasrobbed are we assuming that the throw is only used for development and not for production?

**Comment by getaaron** [comment]
@rlam3 no, there may be keychain errors at any time. Your app should handle failure cases gracefully. Using `try!` is not graceful.

**Comment by rlam3** [comment]
@getaaron Could you please kindly provide an example of a more graceful approach? Thanks!

**Comment by iwasrobbed** [comment]
@rlam3 Look at the `do` `catch` section: https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/ErrorHandling.html

**Comment by getaaron** [comment]
@rlam3 I would be happy to provide a code sample. Can you post a full stack trace and the method you're calling so I can see how you're getting this to occur?
